DSCResources/MSFT_xVMSwitch/MSFT_xVMSwitch.psm1

```diff
@@ -228,48 +228,54 @@ function Test-TargetResource
             # If switch should be present, check the switch type
             if($Ensure -eq 'Present')
             {
-                # If the BandwidthReservsationMode is correct, or if $switch.BandwidthReservationMode is $null which means it isn't supported on the OS
-                Write-Verbose -Message "Checking if Switch $Name has correct BandwidthReservationMode ..."
-                if($switch.BandwidthReservationMode -eq $BandwidthReservationMode -or $switch.BandwidthReservationMode -eq $null)
+                ## Only check the BandwidthReservationMode if specified
+                if($PSBoundParameters.ContainsKey('BandwidthReservationMode'))
```

Comment:
Won't this always return True though since the BandwidthReservationMode parameter defaults to NA? I think you need to remove the default assignment of NA.